### PR TITLE
Verify zk onchain

### LINKED ISSUE #101: Verify zk proofs on chain
Verify zk proofs on chain and remove proof events. Remove proof indexes and stop referencing them in other events (e.g. Attestation).

### PR DESCRIPTION
**Pull request by vivianjeng**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint --check`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Closes #101 

## What is the new behavior?

- Verify proofs before assigning proof index
- 
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/contracts/contracts/Unirep.sol [hunk 1]
```diff
@@ -123,6 +123,10 @@ contract Unirep is IUnirep, zkSNARKHelper, VerifySignature {
             revert AttesterIdNotMatch(attesterId);
     }
 
+    function verifyProof(bool isValid) private pure {
```
Comment (vimwitch) [inline packages/contracts/contracts/Unirep.sol:126]: This doesn't need to be a function.

File: packages/contracts/contracts/Unirep.sol [hunk 2]
```diff
@@ -372,6 +381,8 @@ contract Unirep is IUnirep, zkSNARKHelper, VerifySignature {
     }
 
     /**
+=======
```
Comment (vimwitch) [inline packages/contracts/contracts/Unirep.sol:384]: merge conflict marker

**Review by vimwitch — CHANGES_REQUESTED** [review]

**Review by vimwitch — APPROVED** [review]

File: packages/core/test/utils.ts [hunk 3]
```diff
@@ -127,85 +123,85 @@ const getReputationRecords = async (id: ZkIdentity, userState: UserState) => {
...
-        rep_nullifiers_amount: repNullifiersAmount,
-        selectors: selectors,
-        rep_nonce: nonceList,
-        min_rep: minRep,
-        prove_graffiti: proveGraffiti,
-        graffiti_pre_image: graffitiPreImage,
-    }
-    return stringifyBigInts(circuitInputs)
-}
+// const genReputationCircuitInput = (
```
Comment (vuvoth) [inline packages/core/test/utils.ts:126]: Let remove it if we don't use it

File: packages/core/test/UserState/reputationProofEvent.test.ts [hunk 4]
```diff
@@ -396,12 +400,31 @@ describe('Reputation proof events in Unirep User State', function () {
...
 
             const attesterId = await unirepContract.attesters(attester.address)
-
-            const repProof = await userState.genProveReputationProof(
-                attesterId.toBigInt(),
-                0
+            const reputationRecords = {}
+            reputationRecords[attesterId.toString()] = new Reputation(
+                BigInt(Math.floor(Math.random() * 100) + 10),
+                BigInt(Math.floor(Math.random() * 0)),
```
Comment (vuvoth) [inline packages/core/test/UserState/reputationProofEvent.test.ts:406]: BigInt(Math.floor(Math.random() * 0)) always return 0,right?

**Review by vuvoth — APPROVED** [review]

**Pull request merged**